root/components/ReleaseGroupsList.js

```diff
@@ -0,0 +1,263 @@
+  seriesItemNumbers,
+  showArtists,
+  sortable,
+}: ReleaseGroupsListHeaderProps) => (
+  <thead>
+    <tr>
+      {$c.user_exists && checkboxes ? (
+        <th style={{width: '1em'}}>
+          <input type="checkbox" />
+        </th>
+      ) : null}
+      {seriesItemNumbers ? <th style={{width: '1em'}}>{l('#')}</th> : null}
+      <th className="year c">
+        {sortable
+          ? (
+            <SortableTableHeader
+              label={l('Year')}
+              name="year"
+              order={order}
+            />
+          )
+          : l('Year')}
+      </th>
+      <th>
+        {sortable
+          ? (
+            <SortableTableHeader
+              label={l('Title')}
+              name="name"
+              order={order}
+            />
+          )
+          : l('Title')}
+      </th>
+      {showArtists ? <th className="artist">{l('Artist')}</th> : null}
+      {noGrouping ? (
+        <th>
+          {sortable
+            ? (
+              <SortableTableHeader
+                label={l('Type')}
+                name="primary_type"
+                order={order}
+              />
+            )
+            : l('Type')}
+        </th>
+      ) : null}
+      {noRatings ? null : <th className="rating c">{l('Rating')}</th>}
+      <th className="count c">{l('Releases')}</th>
+    </tr>
+  </thead>
+);
+
+const ReleaseGroupsListEntry = ({
+  $c,
+  checkboxes,
+  index,
+  noGrouping,
+  noRatings,
+  releaseGroup,
+  seriesItemNumbers,
+  showArtists,
+}: ReleaseGroupsListEntryProps) => (
+  <tr className={loopParity(index)} key={releaseGroup.id}>
+    {$c.user_exists && checkboxes ? (
+      <td>
+        <input
+          name={checkboxes}
+          type="checkbox"
+          value={releaseGroup.id}
+        />
+      </td>
+    ) : null}
+    {seriesItemNumbers ? (
+      <td style={{width: '1em'}}>
+        {seriesItemNumbers[releaseGroup.id]}
+      </td>
+    ) : null}
+    <td className="c">
+      {releaseGroup.firstReleaseDate
+        ? new Date(releaseGroup.firstReleaseDate).getFullYear()
+        : '—'}
+    </td>
+    <td>
+      <EntityLink entity={releaseGroup} />
+    </td>
+    {showArtists ? (
+      <td>
+        <ArtistCreditLink
+          artistCredit={releaseGroup.artistCredit}
+        />
+      </td>
+    ) : null}
+    {noGrouping ? (
+      <td>
+        {releaseGroup.typeName
+          // TODO: replace with releaseGroupType when it works here
+          ? releaseGroup.l_type_name
+          : null
+        }
+      </td>
+    ) : null}
+    {noRatings ? null : (
+      <td className="c">
+        <RatingStars entity={releaseGroup} />
+      </td>
+    )}
+    <td className="c">{releaseGroup.release_count}</td>
+  </tr>
+);
+
+const ReleaseGroupsList = ({
+  $c,
+  checkboxes,
+  noGrouping,
+  noRatings,
+  order,
+  releaseGroups,
+  seriesItemNumbers,
+  showArtists,
+  sortable,
+}: ReleaseGroupsListProps) => {
+  const groupedReleaseGroups = groupBy(releaseGroups, 'typeName');
```

Comment:
This actually ensures proper grouping, although it doesn't necessarily ensure proper sorting (say, in RG collection pages). That said, current collection pages don't even group RGs properly, so this already improves on that by making sure there's only one "Album+Compilation" grouping instead of N. We might want to ensure the returned RGs are sorted in the same way they are for artist overview pages, but that's not part of this PR unless you want it to be :) 

BTW, sortable actually sorts globally here, not each mini-table, meaning while the contents of each mini-table will be correctly sorted, the tables *themselves* will also be sorted, so that if the RG with the earliest year of all is "Single" then "Single" will be the first table in the collection if you sort it by year. Not sure if intended.

Reply:
Ah, the previous collection page kept the items sorted globally, so it sorted everything as a single list and then only grouped adjacent items of the same type (so there could be duplicate groups if they weren't adjacent).

The new sorting doesn't seem easier to understand, though grouping them correctly is probably better overall. The UI is just confusing in general (I was expecting the column headers to only sort the specific table they're in, I guess).

Reply:
I guess we could make all sorts of sort sort by type first?

Reply:
Sorting globally is kinda broken, but it is probably less broken than having duplicate groups.

Ideally, we should have a single table with type as a column, and the ability to sort and filter on more than one column. This can wait for website redesign.